x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -248,22 +258,19 @@ func (a *Aggregator) publish(ctx context.Context, period time.Duration) error {
 			entry.histogram.Reset()
 			a.histogramPool.Put(entry.histogram)
 			entry.histogram = nil
```

Comment:
Should we reset/recycle the histogram in the reset method?

Reply:
I didn't do this because the `histogramPool` is defined in the aggregator and resetting + recycling the histogram would mean that a `reset` either is a method on aggregator struct or has a reference to the pool passed to it. Both approaches didn't feel clean. Do you have any suggestions on how to handle this?

Reply:
Hm yeah I see. I think I'd be inclined to pass the pool into `reset`, just like you've done with `newMetricsEntry`. Maybe not the cleanest, but it's at least keeps the create/destroy code similar.

Reply:
Hah, just saw your commit - you beat me to it :)

Reply:
> or has a reference to the pool passed to it

Added a [commit for this](https://github.com/elastic/apm-server/pull/10362/commits/78fdd93390a1efe1a58cf847531e30c3a66ae15c). It doesn't look that bad now 😅, WDYT?

Reply:
Yup, LGTM - thanks.